src/engine/source/builder/policy.hpp

```diff
@@ -0,0 +1,267 @@
+#ifndef _ENVIRONMENT_H
+#define _ENVIRONMENT_H
+
+#include <map>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <fmt/format.h>
+
+#include <json/json.hpp>
+#include <store/istore.hpp>
+
+#include "asset.hpp"
+#include "expression.hpp"
+#include "graph.hpp"
+
+#include "registry.hpp"
+
+namespace builder
+{
+
+constexpr const char* const DECODERS = "decoders";
+constexpr const char* const RULES = "rules";
+constexpr const char* const OUTPUTS = "outputs";
+constexpr const char* const FILTERS = "filters";
+constexpr const char* const INTEGRATIONS = "integrations";
+
+/**
+ * @brief Get the Asset Type object from the string
+ *
+ * @param name
+ * @return Asset::Type
+ * @throws std::runtime_error if the name is not supported
+ */
+Asset::Type getAssetType(const std::string& name);
+
+/**
+ * @brief Intermediate representation of the environment.
```

Comment:
Documentation should also be updated to 'policy'.

Reply:
done